### PR TITLE
Clean up shell code in tests

### PR DESCRIPTION
**Pull request by VladimirSlavik**
Fix shellcheck warnings + organize the scripts a bit.

### CODE REVIEW

**Review by poncovka — APPROVED** [review]
Thanks!

**Review by M4rtinK — APPROVED** [review]
Looks good to me, thanks!

File: .shellcheckrc [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# ShellCheck config file
```
Comment (jkonecny12) [inline .shellcheckrc:1]: Please add this file to the `.structure-config` file.

File: tests/glade_tests.sh [hunk 2]
```diff
@@ -2,13 +2,13 @@
 
 # If $top_srcdir has not been set by automake, import the test environment
 if [ -z "$top_srcdir" ]; then
-    top_srcdir="$(realpath $(dirname "$0")/..)"
-    . ${top_srcdir}/tests/testenv.sh
+    top_srcdir="$(realpath "$(dirname "$0")/..")"
```
Comment (jkonecny12) [inline tests/glade_tests.sh]: Are you sure that this works as expected. The parser will probably not work correctly on all these `"`. Also the original variant is wrong too. I guess the correct way to fix this should be to start escaping the inner quotes.

File: tests/rpm_tests.sh [hunk 3]
```diff
@@ -2,13 +2,13 @@
 
 # If $top_srcdir has not been set by automake, import the test environment
 if [ -z "$top_srcdir" ]; then
-    top_srcdir="$(realpath $(dirname "$0")/..)"
-    . ${top_srcdir}/tests/testenv.sh
+    top_srcdir="$(realpath "$(dirname "$0")/..")"
```
Comment (jkonecny12) [inline tests/rpm_tests.sh]: The same here.

File: tests/unit_tests.sh [hunk 4]
```diff
@@ -2,13 +2,13 @@
 
 # If $top_srcdir has not been set by automake, import the test environment
 if [ -z "$top_srcdir" ]; then
-    top_srcdir="$(realpath $(dirname "$0")/..)"
-    . ${top_srcdir}/tests/testenv.sh
+    top_srcdir="$(realpath "$(dirname "$0")/..")"
```
Comment (jkonecny12) [inline tests/unit_tests.sh]: And here.

File: tests/glade_tests.sh (hunk 2 above)
Comment (jkonecny12) [reply tests/glade_tests.sh]: Although the question is if the `$(..)` sections are not parsed separately.

**Review by jkonecny12 — CHANGES_REQUESTED** [review]
Some things have to be considered, however, we definitely want that file added to the structure file list.

File: tests/glade_tests.sh (hunk 2 above)
Comment (VladimirSlavik) [reply tests/glade_tests.sh]: Yes, it works. As you already guessed, $(...) is another "scope" - it creates a subshell where the quotes are parsed independently of the surroundings. That leads to some really bizarre things like this one here.

File: .shellcheckrc (hunk 1 above)
Comment (VladimirSlavik) [reply .shellcheckrc:1]: Ah, good catch, sorry!

**Comment by VladimirSlavik** [comment]
/kickstart-test --testtype smoke

Update: Added .shellcheckrc to .structure-config

File: tests/glade_tests.sh (hunk 2 above)
Comment (jkonecny12) [reply tests/glade_tests.sh]: Thanks for verification of this.

**Review by jkonecny12 — APPROVED** [review]
Looks good to me now. Thanks!

**Pull request merged**